zerver/lib/topic_mutes.py

```diff
@@ -124,3 +124,20 @@ def is_muted(recipient_id: int, topic: str) -> bool:
         return (recipient_id, topic.lower()) in tups
 
     return is_muted
+
+def get_users_muted_topic_data(topic_name: str, stream_id: int) -> List[Dict[str, Any]]:
+    rows = MutedTopic.objects.filter(
+        stream_id=stream_id,
+        topic_name__iexact=topic_name,
+    ).values(
+        'user_profile',
+        'recipient_id',
+        'date_muted',
+    )
+
+    return [
+        dict(user_profile=UserProfile.objects.get(id=row['user_profile']),
```

Comment:
For better performance, we [avoid doing](https://zulip.readthedocs.io/en/latest/contributing/code-style.html#userprofile-objects-get-client-objects-get-etc) direct `UserProfile.objects.get()` queries, and instead use the `get_user_profile_by_*` functions in `models.py`.

Reply:
Ahh I see, I will be update this, once this pr will recieve more reviews. :)